src/gravel/controllers/orch/orchestrator.py

```diff
@@ -46,25 +46,47 @@ def host_ls(self) -> List[OrchHostListModel]:
         res = self.call(cmd)
         return parse_obj_as(List[OrchHostListModel], res)
 
-    def devices_ls(self) -> List[OrchDevicesPerHostModel]:
-        cmd = {"prefix": "orch device ls"}
+    def devices_ls(
+        self,
+        hostname: Optional[str] = None
+    ) -> List[OrchDevicesPerHostModel]:
+        cmd: Dict[str, Any] = {"prefix": "orch device ls"}
+        if hostname and len(hostname) > 0:
+            cmd["hostname"] = [hostname]
+
         res = self.call(cmd)
         return parse_obj_as(List[OrchDevicesPerHostModel], res)
 
-    def assimilate_all_devices(self) -> None:
+    def assimilate_devices(self, host: str, devices: List[str]) -> None:
+
+        devs: str = ','.join(devices)
         cmd = {
-            "prefix": "orch apply osd",
-            "all_available_devices": True
+            "prefix": "orch daemon add osd",
+            "svc_arg": f"{host}:{devs}"
```

Comment:
this is a synchronous call in the orchestrator ... and will create an 'unmanaged' osd spec

should we pass the devs via a DriveGroupSpec to the async `orch apply osd` command?

Reply:
that explains why it takes so long to finish. I guess the alternative needs to be the spec :\